src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -698,6 +700,17 @@ class LicenceService(
     )
   }
 
+  @Transactional
+  fun removeAPConditions() {
+    val licencesInPSSPeriod = licenceRepository.getAllVariedLicencesInPSSPeriod()
+
+    licencesInPSSPeriod?.forEach { licence ->
+      val licenceWithAPConditions = licence.additionalConditions.filter { LicenceType.valueOf(it.conditionType!!) == LicenceType.AP }
+
+      licenceWithAPConditions?.forEach { this.licenceConditionService.deleteAdditionalCondition(it.licence.id, it.id) }
```

Comment:
Think this will need to be some form of bulk update - otherwise you're doing an additional `n x a` licence loads and deletes where `n` is the number of licences and `a` is the number of additional conditions

Reply:
you could maybe add an overload of `deleteAdditionalCondition`:

```kt
 // old code calls this
  fun deleteAdditionalCondition(licenceId: Long, conditionId: Long) {
    val licenceEntity = licenceRepository
      .findById(licenceId)
      .orElseThrow { EntityNotFoundException("$licenceId") }
    this.deleteAdditionalCondition(licence, listOf(condtionId))
}

//your job calls this directly
 fun deleteAdditionalCondition(licenceId: Licence, conditionIds: List<Long>) {
    // ...modify existing code to filter out many conditions at once ...
}

```

Does that make sense? If not we can have a chat

Reply:
Yes, It makes sense.
I updated the function to n+1 where we will be calling deleteAdditionalConditions once per licence.